diff --git a/ldns/common.h.in b/ldns/common.h.in
index 8bf9654..befd587 100644
--- a/ldns/common.h.in
+++ b/ldns/common.h.in
@@ -37,20 +37,22 @@
  */
 /*@ignore@*/
 /* splint barfs on this construct */
-#ifdef HAVE_STDBOOL_H
-# include <stdbool.h>
-#else
-# ifndef HAVE__BOOL
-#  ifdef __cplusplus
-typedef bool _Bool;
-#  else
-#   define _Bool signed char
+#ifndef __bool_true_false_are_defined
+# ifdef HAVE_STDBOOL_H
+#  include <stdbool.h>
+# else
+#  ifndef HAVE__BOOL
+#   ifdef __cplusplus
+     typedef bool _Bool;
+#   else
+#    define _Bool signed char
+#   endif
 #  endif
+#  define bool _Bool
+#  define false 0
+#  define true 1
+#  define __bool_true_false_are_defined 1
 # endif
-# define bool _Bool
-# define false 0
-# define true 1
-# define __bool_true_false_are_defined 1
 #endif
 /*@end@*/

diff --git a/doc/doxyparse.pl b/doc/doxyparse.pl
index 8bf9654..befd587 100755
@@ -273,7 +295,7 @@ foreach (keys %manpages) {
 
 	print MAN $MAN_MIDDLE;
 
-	if (defined(@$also)) {
+	if (@$also) {
 		print MAN "\n.SH SEE ALSO\n\\fI";
 		print MAN join "\\fR, \\fI", @$also;
 		print MAN "\\fR.\nAnd ";
