diff --git a/ldns/common.h.in b/ldns/common.h.in
index 8bf9654..befd587 100644
--- a/ldns/common.h.in
+++ b/ldns/common.h.in
@@ -37,20 +37,22 @@
  */
 /*@ignore@*/
 /* splint barfs on this construct */
-#ifdef HAVE_STDBOOL_H
-# include <stdbool.h>
-#else
-# ifndef HAVE__BOOL
-#  ifdef __cplusplus
-typedef bool _Bool;
-#  else
-#   define _Bool signed char
+#ifndef __bool_true_false_are_defined
+# ifdef HAVE_STDBOOL_H
+#  include <stdbool.h>
+# else
+#  ifndef HAVE__BOOL
+#   ifdef __cplusplus
+     typedef bool _Bool;
+#   else
+#    define _Bool signed char
+#   endif
 #  endif
+#  define bool _Bool
+#  define false 0
+#  define true 1
+#  define __bool_true_false_are_defined 1
 # endif
-# define bool _Bool
-# define false 0
-# define true 1
-# define __bool_true_false_are_defined 1
 #endif
 /*@end@*/
 
